openapi2/openapi2.go

```diff
@@ -100,6 +103,54 @@ type (
 	}
 )
 
+func (o *Operation) toMap() (map[string]interface{}, error) {
+	type Alias Operation
+	data, err := json.Marshal((*Alias)(o))
+	if err != nil {
+		return nil, fmt.Errorf("json marshal: %v", err)
+	}
+
+	m := map[string]interface{}{}
+	if err := json.Unmarshal(data, &m); err != nil {
+		return nil, fmt.Errorf("json unmarshal: %v", err)
+	}
+
+	if o.Extend != nil {
+		if err := mergo.Merge(&m, o.Extend, mergo.WithOverride); err != nil {
+			return nil, fmt.Errorf("merge extend: %v", err)
+		}
+	}
+	return m, nil
+}
+
+// MarshalJSON implements the json.Marshaller interface.
+func (o *Operation) MarshalJSON() ([]byte, error) {
+	if o.Extend == nil {
+		type Alias Operation
+		return json.Marshal((*Alias)(o))
+	}
+
+	m, err := o.toMap()
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(m)
+}
+
+// MarshalYAML implements the yaml.Marshaller interface.
+func (o *Operation) MarshalYAML() (interface{}, error) {
```

Comment:
Might worth double check this one. With this return type, doesn't seem to be properly marshaling anything.

Reply:
Have you got an example? It's working on the tests https://godoc.org/gopkg.in/yaml.v2#Marshaler